core/transactions.hxx

```diff
@@ -127,21 +129,49 @@ using txn_complete_callback =
 class transactions : public couchbase::transactions::transactions
 {
   public:
+    /**
+     * @internal
+     * Called internally
+     */
+    transactions(core::cluster cluster, const couchbase::transactions::transactions_config::built& config);
+    transactions(core::cluster cluster, const couchbase::transactions::transactions_config& config);
+
+    /**
+     * @brief Destructor
+     */
+    ~transactions();
+
     /**
      * @brief Create a transactions object.
      *
      * Creates a transactions object, which can be used to run transactions within the current thread.
      *
      * @param cluster The cluster to use for the transactions.
      * @param config The configuration parameters to use for the transactions.
+     * @param cb Called when the transactions object has been created or an error has occurred during the creation.
      */
-    transactions(core::cluster cluster, const couchbase::transactions::transactions_config::built& config);
-    transactions(core::cluster cluster, const couchbase::transactions::transactions_config& config);
+    static void create(core::cluster cluster,
+                       const couchbase::transactions::transactions_config::built&,
+                       utils::movable_function<void(std::error_code, std::shared_ptr<transactions>)>&& cb);
```

Comment:
@thejcfactor Does returning a shared pointer here work for Python/Node?

Reply:
the wrapper could always use future/promise to wait for completion

Reply:
I have tested out the changes w/o any problems.  When we create the `transactions` obj we do it in a blocking fashion so I think using one of the overloads that returns the `std::future<std::pair<>>` is the path we would take. Either way, we can get to correctly raising an exception is there is an issue w/ opening the metadata bucket. 👍